### PR TITLE
bump Microsoft/hcsshim to v0.12.5: use containerd/cgroup/v3

### PR DESCRIPTION
**Pull request by pacoxu**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:
Fixes None

#### Special notes for your reviewer:
When I learn containerd/cgroup v2 supports, I find we are still using `github.com/containerd/cgroups` http://github.com/containerd/cgroups v1.1.0.

- Containerd is using：[github.com/containerd/cgroups/v3](http://github.com/containerd/cgroups/v3) v3.0.3`

But in Kube, `Microsoft/hcsshim 0.8` is using `github.com/containerd/cgroups`.
- Microsoft/hcsshim to v0.12.5 uses containerd/cgroup/v3 after https://github.com/microsoft/hcsshim/pull/1706([v0.10.0](https://github.com/microsoft/hcsshim/releases/tag/v0.10.0))

#### Does this PR introduce a user-facing change?
```release-note
None
```

### CODE REVIEW

**Comment by pacoxu** [comment]
for dependencies change
```
-Direct Dependencies: 187 
+Direct Dependencies: 188 

-Transitive Dependencies: 327 
+Transitive Dependencies: 366 

-Total Dependencies: 367 
+Total Dependencies: 405 
 Max Depth Of Dependencies: 25 
```

for unwanted dependencies update
```
     "github.com/gogo/protobuf": [
-        "github.com/containerd/cgroups",
-        "github.com/containerd/typeurl",

      "github.com/pkg/errors": [
-        "github.com/containerd/continuity",
-        "github.com/containerd/fifo",
-        "github.com/containerd/go-runc",
-        "github.com/containerd/typeurl",
-         "gotest.tools/v3",

      "golang.org/x/exp": [
+        "github.com/Microsoft/hcsshim",
+        "github.com/cilium/ebpf",

      "google.golang.org/genproto": [
-        "github.com/containerd/ttrpc",
+        "github.com/Microsoft/hcsshim",
```

**Comment by pacoxu** [comment]
/cc @liggitt @dims 
for dep update

File: hack/unwanted-dependencies.json [hunk 1]
```diff
@@ -181,15 +174,17 @@
         "github.com/Microsoft/hcsshim"
       ],
       "golang.org/x/exp": [
+        "github.com/Microsoft/hcsshim",
         "github.com/antlr4-go/antlr/v4",
+        "github.com/cilium/ebpf",
```
Comment (liggitt) [inline hack/unwanted-dependencies.json:179]: adding more deps on an explicitly unstable library is *really* unfortunate ... can we determine what experimental bits they are using and proposal upstream changes to drop them out?

File: hack/unwanted-dependencies.json [hunk 2]
```diff
@@ -181,15 +174,17 @@
...
         "github.com/google/cel-go",
         "k8s.io/mount-utils"
       ],
       "google.golang.org/appengine": [
         "github.com/prometheus/client_golang"
       ],
       "google.golang.org/genproto": [
-        "github.com/containerd/ttrpc",
+        "github.com/Microsoft/hcsshim",
```
Comment (liggitt) [inline hack/unwanted-dependencies.json:187]: I'm really surprised to see something new pick up a dep here... we're trying *really* hard to drop this dep entirely (https://github.com/kubernetes/kubernetes/issues/113366)

We've already carved out all the generic proto libs that attracted use... what is hcsshim using in the root genproto dep? can we get that updated to no longer have this dep?

**Comment by liggitt** [comment]
this brought in a *lot* of new deps... can we figure out why and if we can help a specific upstream isolate or drop them to avoid that?

**Comment by dims** [comment]
> this brought in a lot of new deps (38 new transitive deps)... can we figure out why and if we can help a specific upstream isolate or drop them to avoid that?

cc @kiashok

**Comment by pacoxu** [comment]
> > this brought in a lot of new deps (38 new transitive deps)... can we figure out why and if we can help a specific upstream isolate or drop them to avoid that?
> 
> cc @kiashok
> 
> Note from Kirtana before ... `We have a plan to move HNS pkg out of hcsshim to its own new repo before the Oct/Nov k8s release cycle`
> 
> When we do get this split for HNS, we won't need the entire hcsshim and it will help drop the transitive numbers as well.

Good news. We may wait for that.

**Comment by pacoxu** [comment]
For ebpf usage in runc, Kir has some explainations in https://github.com/kubernetes/kubernetes/pull/117457#issuecomment-1528070248.

File: hack/unwanted-dependencies.json (hunk 1 above)
Comment (pacoxu) [reply hack/unwanted-dependencies.json:179]: https://github.com/cilium/ebpf/issues/1095 
We have opened an issue for this before.

**Pull request closed without merging**